### PR TITLE
feat: function to trim prefixes

### PR DESCRIPTION
**Pull request by ashishbhate**

### CODE REVIEW

File: string.go [hunk 1]
```diff
@@ -43,6 +43,17 @@ func TrimSuffixes(s string, suffixes ...string) string {
 	return s
 }
 
+// TrimPrefixes returns s without any of the provided trailing prefix strings.
+func TrimPrefixes(s string, prefixes ...string) string {
+	s = strings.TrimSpace(s)
+	for _, prefix := range prefixes {
+		if strings.HasPrefix(s, prefix) {
```
Comment (ferhatelmas) [inline string.go]: no need for check, just trim

File: string.go [hunk 2]
```diff
@@ -43,6 +43,17 @@ func TrimSuffixes(s string, suffixes ...string) string {
 	return s
 }
 
+// TrimPrefixes returns s without any of the provided trailing prefix strings.
```
Comment (ferhatelmas) [inline string.go]: trailing or leading ?

File: string_test.go [hunk 3]
```diff
@@ -467,3 +467,22 @@ func TestMap(t *testing.T) {
...
+		input, expected string
+		prefixes        []string
+	}{
+		{input: "", expected: "", prefixes: nil},
+		{input: "", expected: "", prefixes: []string{"a"}},
+		{input: "   b   ", expected: "", prefixes: []string{"b"}},
+		{input: "   c", expected: "", prefixes: []string{"c"}},
+		{input: "   d\f", expected: "", prefixes: []string{"d"}},
+		{input: "   d\fe\n", expected: "", prefixes: []string{"d\fe"}},
+		{input: "   good food", expected: "food", prefixes: []string{"good"}},
```
Comment (ferhatelmas) [inline string_test.go]: could you add with multiple prefix test ? it seems it can be multiple but not used.

File: string.go [hunk 4]
```diff
@@ -43,6 +43,17 @@ func TrimSuffixes(s string, suffixes ...string) string {
...
+func TrimPrefixes(s string, prefixes ...string) string {
```
Comment (ferhatelmas) [inline string.go]: I expect to trim all with this name but we stop at the first one. 

I'm not sure if I get the goal of this function, where is it needed ?

**Comment by ashishbhate** [comment]
After discussion. Closed in favour of a more comprehensive solution coming up in another PR.

**Pull request closed without merging**